test/configCases/externals/object-externals-root/webpack.config.js

```diff
@@ -0,0 +1,26 @@
+/** @type {import("../../../../").Configuration} */
+module.exports = {
+	target: ["web", "es2020"],
+	entry: {
+		main: "./index",
+		other: "./other"
+	},
+	externalsType: {
+		"static-import": "commonjs",
+		"dynamic-import": "import",
+		fallback: "window"
```

Comment:
Can we add a test case with `amd`, also we support `system` and other

To be honestly we can use https://github.com/webpack/webpack/blob/main/lib/dependencies/CommonJsFullRequireDependency.js#L93 and https://github.com/webpack/webpack/blob/main/lib/dependencies/ImportDependency.js#L40 

Do I understand correctly that we are trying to provide the opportunity to use one or another type external depending by dependency category?

Maybe we can introduce new function to our dependencies

Reply:
> Can we add a test case with amd, also we support system and other

Sure.

> Do I understand correctly that we are trying to provide the opportunity to use one or another type external depending by dependency category?

The original requirement is that when we build for NPM package, we need to keep `import()` as `import()` and convert `import` to `commonjs` (similar to https://github.com/webpack/webpack/discussions/16272). Considering that we previously added the type of `'module-import'`, supporting map types for `externalsTypes` would be more scalable and more convenient for users. We had thought about exposing some fields of dependency in a function before, but this would expose internal abstractions, so we need to categorize possible categories for users in advance. The finer the category judgment here, the better it can prevent us from having to split existing categories in the future.

> Maybe we can introduce new functions into our dependencies.
I need to review all possible external types again, I previously thought that category covered all enumerations, but now we need to consider the field type.

Reply:
> Can we add a test case with amd, also we support system and other

I added a test case for `amd`. But I am not clear of the `system` here, does Systemjs got a dedicated import method?